endpoints/events/vtrack.go

```diff
@@ -269,25 +270,34 @@ func getAccountId(httpRequest *http.Request) string {
 	return httpRequest.URL.Query().Get(AccountParameter)
 }
 
-// modifyVastXml modifies BidCacheRequest element Vast XML data
-func modifyVastXml(externalUrl string, data json.RawMessage, bidid string, bidder string, accountId string, timestamp int64) json.RawMessage {
-	c := string(data)
+// modifyVastXmlString rewrites and returns the string vastXML or an empty string if it was left unmodified
+func ModifyVastXmlString(externalUrl string, c string, bidid string, bidder string, accountId string, timestamp int64) string {
 	ci := strings.Index(c, ImpressionCloseTag)
 
 	// no impression tag - pass it as it is
 	if ci == -1 {
-		return data
+		return ""
 	}
 
 	vastUrlTracking := GetVastUrlTracking(externalUrl, bidid, bidder, accountId, timestamp)
 	impressionUrl := "<![CDATA[" + vastUrlTracking + "]]>"
 	oi := strings.Index(c, ImpressionOpenTag)
 
 	if ci-oi == len(ImpressionOpenTag) {
-		return json.RawMessage(strings.Replace(c, ImpressionOpenTag, ImpressionOpenTag+impressionUrl, 1))
+		return strings.Replace(c, ImpressionOpenTag, ImpressionOpenTag+impressionUrl, 1)
 	}
 
-	return json.RawMessage(strings.Replace(c, ImpressionCloseTag, ImpressionCloseTag+ImpressionOpenTag+impressionUrl+ImpressionCloseTag, 1))
+	return strings.Replace(c, ImpressionCloseTag, ImpressionCloseTag+ImpressionOpenTag+impressionUrl+ImpressionCloseTag, 1)
+}
+
+// modifyVastXml modifies BidCacheRequest element Vast XML data
+func ModifyVastXmlJSON(externalUrl string, data json.RawMessage, bidid string, bidder string, accountId string, timestamp int64) json.RawMessage {
+	c := string(data) // FIXME: should decode json, or escapes like "\u003e" will not be recognized as "<"
```

Comment:
`FIXME` Ah! Can you fix it before merging it in please?

Reply:
I was debating fixing this as a case of doing it fast vs doing it right, since only the Go serializer insists on escaping html - but perhaps I should.

Reply:
vscode nags me about the naming of these functions. It says they should be `ModifyVastXMLJSON` and `ModifyVastXMLString` respectively (probably because it doesn't know VAST is also an acro). Thoughts ?

Reply:
I don't have a strong opinion on the name. You're right. It should technically be called ModifyVASTXMLJSON.. but I mean... my eyes! I would just leave it as is and not worry.

Could you please fill me in on why escaping the HTML is a problem here? I just don't have the context. Anyway, you can customize the Go json encoder to turn this feature off:
https://play.golang.org/p/C79fK-_OKe5

Reply:
Sorry, html escapes in Go are a red herring here and probably worth a separate discussion.
Regarding the issue at hand, PBS has no control over how adm was encoded, so it has to be able to handle it, so I replaced `string()` with a proper `UnMarshal()`